docs/source/point-evaluation.rst

```diff
@@ -98,6 +100,235 @@ in parallel. There is no special API, but there are some restrictions:
 
 
+Primary API: Interpolation onto a vertex-only mesh
+--------------------------------------------------
+
+Firedrake's principal API for evaluating functions at arbitrary points,
+interpolation onto a :func:`~.VertexOnlyMesh`, is designed for evaluating a
+function at many points, or repeatedly, and for creating expressions which
+contain point evaluations. It is parallel-safe. Whilst :meth:`~.Function.at`
+produces a list of values, cross-mesh interpolation onto
+:func:`~.VertexOnlyMesh` gives Firedrake :py:class:`~.Function`\s.
+
+This is discussed in detail in :cite:`nixonhill2023consistent` but, briefly,
+the idea is that the :func:`~.VertexOnlyMesh` is a mesh that represents a
+point cloud domain. Each cell of the mesh is a vertex at a chosen location in
+space. As usual for a mesh, we represent values by creating functions in
+function spaces on it. The only function space that makes sense for a mesh
+whose cells are vertices is the space of piecewise constant functions, also
+known as the Polynomial degree 0 Discontinuous Galerkin (P0DG) space.
+
+Our vertex-only meshes are immersed in some 'parent' mesh. We perform point
+evaluation of a function :math:`f` defined in a function space
+:math:`V` on the parent mesh by interpolating into the P0DG space on the
+:func:`~.VertexOnlyMesh`. For example:
+
+.. literalinclude:: ../../tests/vertexonly/test_vertex_only_manual.py
+   :language: python3
+   :dedent:
+   :lines: 7-26
+
+will print ``[0.02, 0.08, 0.18]``, the values of :math:`x^2 + y^2` at the
+points :math:`(0.1, 0.1)`, :math:`(0.2, 0.2)` and :math:`(0.3, 0.3)`.
+
+Note that ``f_at_points`` is a :py:class:`~.Function` which takes
+on *all* the values of ``f`` evaluated at ``points``. The cell ordering of a
+:func:`~.VertexOnlyMesh` follows the ordering of the list of points it is given
+at construction. In general :func:`~.VertexOnlyMesh` accepts any numpy array of
+shape ``(num_points, point_dim)`` (or equivalent list) as the set of points to
+create disconnected vertices at.
+
+The operator for evaluation at the points specified can be
+created by making an :py:class:`~.Interpolator` acting on a
+:py:func:`~.TestFunction`
+
+.. code-block:: python3
+
+   u = TestFunction(V)
+   Interpolator(u, P0DG)
+
+For more on :py:class:`~.Interpolator`\s and interpolation see the
+:doc:`interpolation <interpolation>` section.
+
+
+Vector and tensor valued function spaces
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When interpolating from vector or tensor valued function spaces, the P0DG
+function space on the vertex-only mesh must be a
+:py:func:`~.VectorFunctionSpace` or :py:func:`~.TensorFunctionSpace`
+respectively. For example:
+
+.. code-block:: python3
+
+   V = VectorFunctionSpace(parent_mesh, "CG", 2)
+
+or
+
+.. code-block:: python3
+
+   V = FunctionSpace(parent_mesh, "N1curl", 2)
+
+each require
+
+.. code-block:: python3
+
+   vom = VertexOnlyMesh(parent_mesh, points)
+   P0DG_vec = VectorFunctionSpace(vom, "DG", 0)
```

Comment:
Do you not need to provide `dim` as an argument?

Reply:
For `VectorFunctionSpace`? Nope. From the `VectorFunctionSpace` docs
```
    :arg dim: An optional number of degrees of freedom per function
       space node (defaults to the geometric dimension of the mesh).
```

Reply:
Yes of course.